py/desitarget/targets.py

```diff
@@ -387,6 +390,90 @@ def initial_priority_numobs(targets, scnd=False,
     return outpriority, outnumobs
 
 
+def calc_numobs_more(targets, zcat, obscon):
+    """
+    Calculate target NUMOBS_MORE from masks, observation/redshift status.
+
+    Parameters
+    ----------
+    targets : :class:`~numpy.ndarray`
+        numpy structured array or astropy Table of targets. Must include
+        the columns `DESI_TARGET`, `BGS_TARGET`, `MWS_TARGET`
+        (or their SV/cmx equivalents).
+    zcat : :class:`~numpy.ndarray`
+        numpy structured array or Table of redshift information. Must
+        include 'Z', `ZWARN`, `NUMOBS` and be the same length as
+        `targets`. May also contain `NUMOBS_MORE` if this isn't the
+        first time through MTL and `NUMOBS > 0`.
+    obscon : :class:`str`
+        A combination of strings that are in the desitarget bitmask yaml
+        file (specifically in `desitarget.targetmask.obsconditions`), e.g.
+        "DARK|GRAY". Governs the behavior of how priorities are set based
+        on "obsconditions" in the desitarget bitmask yaml file.
+
+    Returns
+    -------
+    :class:`~numpy.array`
+        Integer array of number of additional observations (NUMOBS_MORE).
+
+    Notes
+    -----
+        - Will automatically detect if the passed targets are main
+          survey, commissioning or SV and behave accordingly.
+        - Most targets are updated to NUMOBS_MORE = NUMOBS_INIT-NUMOBS.
+          Special cases include BGS targets which always get NUMOBS_MORE
+          of 1 in bright time and QSO "tracer" targets which always get
+          NUMOBS_MORE=0 in dark time.
+    """
+    # ADM check the input arrays are the same length.
+    assert len(targets) == len(zcat)
```

Comment:
The function assumes that the items in `targets` and `zcat` come in the same order. There has to be some check on that internal to the function (just in case this function ends up being used outside `make_mtl`)

Reply:
This is an excellent point; thanks for catching this. I've updated the `assert` statement to check that the `targets` and `zcat` catalogs have the same order on `TARGETID`. This was _such_ a good point, that it made me also realize that we need the same check in the `targets.calc_priority()` function, so I added it there, too!